### PR TITLE
Namespace the run_state variable

### LINKED ISSUE #285: node.run_state['root_path'] conflicts with pyenv cookbook
## :ghost: Brief Description

When using both the rbenv and pyenv cookbooks together and installing python and ruby versions for the same user, the resources from the 2nd cookbook do not work. This is because the same internal run_state attribute is used for tracking the root_path in both cookbooks.

## :pancakes: Cookbook version

2.4.0

## :woman_cook:  Chef-Infra Version

14.13.11

## :tophat: Platform details

RHEL7

## Steps To Reproduce

1. Write cookbook that creates a user and installs a ruby and python version for that user:
  ```ruby
  user 'jenkins' do
    manage_home true
  end

  rbenv_user_install 'jenkins'

  rbenv_ruby '2.7.1' do
    user 'jenkins'
  end

  pyenv_user_install 'jenkins'

  pyenv_python '3.8.2' do
    user 'jenkins'
  end
```
2. Run the recipe against a node in test-kitchen or a test env.
3. See the pyenv python installation fail

## :police_car: Expected behavior

I expect that both the ruby and python versions are installed with no error.

## :heavy_plus_sign: Additional context

I can work around the problem with a ruby block in between the pyenv and rbenv resources:
```ruby
ruby_block 'unset run_state' do
  block do
    node.run_state['root_path'] = {}
  end
end
```

### PR DESCRIPTION
**Pull request by petracvv**
Signed-off-by: Mihai Petracovici <petracvv@users.noreply.github.com>

# Description
This namespaces the run_state variables used internally by the cookbook to avoid conflicts with other cookbooks.

## Issues Resolved

#285 

## Check List

- [ ] All tests pass. See TESTING.md for details.
- [x] New functionality includes testing.
- [x] New functionality has been documented in the README if applicable.

### CODE REVIEW

**Comment by petracvv** [comment]
Looks like tests are failing because the unit tests are not in a spec directory. Is that something I should fix?

File: libraries/helpers.rb [hunk 1]
```diff
@@ -23,12 +23,14 @@ class Chef
...
     module Helpers
       def self.root_path(node, resource_user = nil)
-        node.run_state['root_path'] ||= {}
+        node.run_state['sous-chefs'] ||= {}
+        node.run_state['sous-chefs']['ruby_rbenv'] ||= {}
+        node.run_state['sous-chefs']['ruby_rbenv']['root_path'] ||= {}
 
         if resource_user
-          node.run_state['root_path'][resource_user]
+          node.run_state['sous-chefs']['ruby_rbenv']['root_path'][resource_user]
```
Comment (damacus) [inline libraries/helpers.rb:31]:
```suggestion
          node.run_state['ruby_rbenv']['root_path'][resource_user]
```

Just using ruby_rbenv is probably enough here considering we have the `ruby_rbenv` namespace on supermarket

**Review by damacus — CHANGES_REQUESTED** [review]
Just using ruby_rbenv is probably enough here considering we have the ruby_rbenv namespace on supermarket

**Comment by petracvv** [comment]
I think to be consistent with the pyenv cookbook, we should leave it under the sous-chefs namespace (see sous-chefs/pyenv#70)

**Comment by kitchen-porter** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Please include a CHANGELOG entry.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This Pull Request is probably missing tests.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by xorima** [comment]
Hey @petracvv 

Can we get a changelog entry so this can be shipped please?

Thanks

**Comment by kitchen-porter** [comment]
Released as: [3.0.0](https://github.com/sous-chefs/ruby_rbenv/releases/tag/3.0.0)

**Comment by petracvv** [comment]
Thanks for the merge! Sorry about taking so long to finish this

**Pull request merged**